vowpalwabbit/core/src/reductions/automl.cc

```diff
@@ -92,14 +92,66 @@ void finish_example(VW::workspace& all, automl<CMType>& data, VW::multi_ex& ec)
   VW::finish_example(all, ec);
 }
 
+template <typename CMType>
+void output_target_model(VW::workspace& all, automl<CMType>& data)
+{
+  options_i& options = *all.options;
+  if (!options.was_supplied("predict_only_model")) { return; }
+  // Clear non-champ weights first
+
+  std::swap(*data.cm->_gd_normalized, data.cm->per_live_model_state_double[0]);
+  std::swap(*data.cm->_gd_total_weight, data.cm->per_live_model_state_double[1]);
+  std::swap(*data.cm->_sd_gravity, data.cm->per_live_model_state_double[2]);
+  std::swap(*data.cm->_cb_adf_event_sum, data.cm->per_live_model_state_uint64[0]);
+  std::swap(*data.cm->_cb_adf_action_sum, data.cm->per_live_model_state_uint64[1]);
+
+  // Clear all non-champ weights
+  clear_non_champ_weights(data.cm->weights, data.cm->estimators.size(), data.cm->wpp);
+
+  // Adjust champ weights to new single-model space
+  data.cm->weights.adjust_weights_single_model(data.cm->wpp, 0);
+
+  std::set<std::string> aml_opts = {"automl", "global_lease", "cm_type", "priority_type", "priority_challengers",
```

Comment:
add assert in the setup func of automl ensuring the count stays constant with a comment saying you have to update this bit over here and they always have to match to avoid future regressions.

Reply:
I was actually able to get rid of this by looking at the option group instead